### PR TITLE
Import German translation from wangqr

### PR DESCRIPTION
**Pull request by TheOneric**
The current meson integration for POT generation is incomplete and misses out a lot of strings. The last commit improves this for strings from C++ source files, but strings from other sources are still missing. For `*.xml` and `*.desktop` adding support might be as simple as renaming to `.in.in` instead of `.template.in` and dropping any extra underscores added for `intltool`. ~~Re `packages/desktop/aegisub.appdata.xml.template.in` also see \[*issue creation pending*\].~~
For the remaining formats I have no idea what the best way to integrate them is.

For this PR’s POT and PO update, I therefore stuck with `make_pot.sh` and `msgmerge`. If you’d prefer keeping the changeset down, I can omit rebasing non-`de` PO files to the new POT version.

Cosmetic nit: it might be a good idea to unify the (original) message texts for `ffms2` and `bestsource`’s keyframe and timecode indexing/scanning dialogue.

### CODE REVIEW

**Comment by TheOneric** [comment]
Found a way to make `aegisub.appdata.xml` build with `i18n` afterall and also discovered the existing `.desktop` integration is borked. The latter also affects the POT update here, so converted to draft. I’ll fix this up and add the desktop and appdata stuuf to this PR later.

**Comment by TheOneric** [comment]
Fixed desktop file string extraction and added meson integration for desktop and appdata files + updating obsolete links in `aegisub.appdata.xml`. Should be ready for review.

Remaining future improvements for someone more familiar with meson i18n details and the build setup:
 - automatically replace release version and date in `packages/desktop/aegisub.appdata.in.in`
 - somehow wire up remaining translatable string sources to meson’s `aegisub-pot` target, so it can supersede `make_pot.sh` . This will require custom logic. See comment in `po/meson.build` and `make_pot.sh` for details.

File: po/de.po [hunk 1]
```diff
@@ -1955,6 +1981,7 @@ msgstr "Ändere Anfang"
 #: ../src/command/time.cpp:119
 msgid "Change start times of lines to the previous line's end time"
 msgstr ""
+"Setze den Startzeit der Zeilen auf das Ende der jeweiligen Vorgängerzeile"
```
Comment (arch1t3cht) [inline po/de.po:1984]: `den Startzeit` -> `die Startzeit`

File: po/de.po [hunk 2]
```diff
@@ -6365,33 +6369,39 @@ msgstr ""
...
 
 #: ../src/preferences.cpp:501
 msgid ""
 "This script will be executed to load video files that aren't\n"
 "VapourSynth scripts (i.e. end in .py or .vpy).\n"
 "The filename variable stores the path to the file."
 msgstr ""
+"Dieses Skript wird ausgeführt um Videodateien zu laden,\n"
+"die nicht bereits VapourSynth-Skripte sind (.py- oder .vpy-Endung).\n"
+"Die Variable speichert den Dateipfad."
```
Comment (arch1t3cht) [inline po/de.po:6391]: In the original message, `filename` is the name of the variable in the Python script. So the third line should be something like `Die Variable filename speichert den Dateipfad.`, possibly also with quotes around it.

Same for the audio script message at `preferences.cpp:510`

File: po/de.po [hunk 3]
```diff
@@ -6821,21 +6831,23 @@ msgstr "Aktuelle Framezeit und -nummer"
...
 
 #: ../src/video_provider_vs.cpp:137
 msgid "Executing VapourSynth Script"
-msgstr ""
+msgstr "VapourSynth-Skript wird ausgeführt"
 
 #: ../src/video_provider_vs.cpp:147
 msgid "Failed to execute script! Press \"Close\" to continue."
-msgstr ""
+msgstr "Skriptausführung fehlgeschlagen! Drücke \"Schließen\" um fortzufahren."
```
Comment (arch1t3cht) [inline po/de.po:6850]: Should this be `Drücken Sie` to be consistent with previous translations? Though I can't say I know the convention for these kinds of phrases and I can't find a precedent in the existing .po file.

**Comment by arch1t3cht** [comment]
Thanks a lot! This mostly LGTM except for a few corrections on the added German translations.

I did also update some of the bestsource messages (actually this was long overdue since some of them were carried over from back when it was still a proof-of-concept and never updated) in 0c057ebddb1b28780a1d70bbc9f6d6d48a25f554, though I still kept some of them. I replaced "Exacting" with "Indexing" and "Finding Keyframes and Timecodes..." with "Reading timecodes and frame/sample data" (this isn't strictly accurate since bestsource doesn't read samples here, but it probably still makes the most sense for the user), but I kept "Scanning" as it is since that is a separate step that (for now) needs to run every time the file is opened, as opposed to the Indexing step that only needs to happen once per file.

**Comment by TheOneric** [comment]
Thanks for the review! Translations are now fixed as suggested and rebased for the new BS source strings.
To make backporting to wangqr easier I split fixing existing translations from wangqr into a separate commit. As a bonus I found and fixed another mistake in the old 2014 translations (`Wahlen` → `Wählen`).

New version now also includes other `make_pot.sh` improvements from wangqr and some new additional improvements. Some noteworthy points:
  - Using gettext for appdata added the developer name `Aegisub group` as a translateable string. I’m unsure if this should be treated as a proper noun and left as is, or localised. (`Aegisub Group` does suggest a higher degree of organisation and unity then actually exists in reality.)
  - Prior to gettext 0.21 *(July 2020)*, msgmerge would [reverse the order of commented strings](https://savannah.gnu.org/bugs/?58778). wangqr’s version of `make_pot.sh` always runs msgmerge twice for each updated language to workaround this issue. However, this workaround was added when there was no released version with a fix yet. At this point we might not really need to add this workaround here as well.
  
New changes which can be backported to wangqr/Aegisub afterwards:
 - update appdata links
 - use gettext for appdata extraction (and merging; requires droppping underscores)
 - new fixes for German translation
 - proper shell portability

**Comment by arch1t3cht** [comment]
Thanks again! There was one more missing space in the VapourSynth description text, but I just fixed that myself.

**Pull request merged**